plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1755,226 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
                 .filter(path -> path.contains("/" + CHANGE_DATA_FOLDER_NAME))
                 .collect(toImmutableSet());
     }
+
+    @Test
+    public void testPartitionFilterQueryNotDemanded()
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_filter_not_demanded",
+                "(x VARCHAR, part VARCHAR) WITH (PARTITIONED_BY = ARRAY['part'])",
+                ImmutableList.of("'a', 'part_a'", "'b', 'part_b'"))) {
+            assertQuery("SELECT * FROM %s WHERE x='a'".formatted(table.getName()), "VALUES('a', 'part_a')");
+            assertQuery("SELECT * FROM %s WHERE part='part_a'".formatted(table.getName()), "VALUES('a', 'part_a')");
+        }
+    }
+
+    @Test
+    public void testQueryWithoutPartitionOnNonPartitionedTableNotDemanded()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+
+        try (TestTable table = new TestTable(
+                new TrinoSqlExecutor(getQueryRunner(), session),
+                "test_no_partition_table_",
+                "(x VARCHAR, part VARCHAR)",
+                ImmutableList.of("('a', 'part_a')", "('b', 'part_b')"))) {
+            assertQuery("SELECT * FROM %s WHERE x='a'".formatted(table.getName()), "VALUES('a', 'part_a')");
+            assertQuery("SELECT * FROM %s WHERE part='part_a'".formatted(table.getName()), "VALUES('a', 'part_a')");
```

Comment:
`session` argument looks missing.

Reply:
missed on purpose, added one explaining assertion that the property is not set

Reply:
Then, I don't understand this test's purpose. Why are you specifying `session` when creating a table?

Reply:
fixed

Reply:
Where's the test case ensuring query success on non-partitioned tables with `query_partition_filter_required` enabled?

Reply:
Oh, now I get it.